src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsWriter.cs

```diff
@@ -0,0 +1,174 @@
+        /// <param name="ct">Cancellation token</param>
+        public ValueTask WriteUnsignedLongAsync(ulong v, bool isAsync, CancellationToken ct) => WriteLongAsync((long)v, isAsync, ct);
+
+        /// <summary>
+        /// Writes float value to out buffer, as little-endian.
+        /// </summary>
+        /// <param name="v">Value to write</param>
+        /// <param name="isAsync">Whether caller method is executing asynchronously.</param>
+        /// <param name="ct">Cancellation token</param>
+        public async ValueTask WriteFloatAsync(float v, bool isAsync, CancellationToken ct)
+        {
+            Debug.Assert(!float.IsInfinity(v) && !float.IsNaN(v), "Float value is out of range.");
+
+            var len = sizeof(float);
+            BinaryPrimitives.WriteInt32LittleEndian(GetBuffer(len), BitConverter.SingleToInt32Bits(v));
+            await WriteBytesAsync(GetBuffer(len), 0, len, isAsync, ct).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Writes double value to out buffer, as little-endian.
+        /// </summary>
+        /// <param name="v">Value to write</param>
+        /// <param name="isAsync">Whether caller method is executing asynchronously.</param>
+        /// <param name="ct">Cancellation token</param>
+        public async ValueTask WriteDoubleAsync(double v, bool isAsync, CancellationToken ct)
+        {
+            Debug.Assert(!double.IsInfinity(v) && !double.IsNaN(v), "Double value is out of range.");
+
+            var len = sizeof(double);
+            BinaryPrimitives.WriteInt64LittleEndian(GetBuffer(len), BitConverter.DoubleToInt64Bits(v));
+            await WriteBytesAsync(GetBuffer(len), 0, len, isAsync, ct).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Writes partial long value to out buffer, as little-endian. This should be ideally used for writing date, 
+        /// time and related data types. The length specified should be between 0 to 8 bytes supported by long.
+        /// </summary>
+        /// <param name="v">Value to write</param>
+        /// <param name="length">Length to fill</param>
+        /// <param name="isAsync">Whether caller method is executing asynchronously.</param>
+        /// <param name="ct">Cancellation token</param>
+        public async ValueTask WritePartialLongAsync(long v, int length, bool isAsync, CancellationToken ct)
+        {
+            Debug.Assert(length >= 0, "Length should not be negative");
+            Debug.Assert(length <= 8, "Length specified is longer than the size of a long");
+
+            var len = sizeof(long);
+            var buffer = GetBuffer(len);
+            for (int i = 0; i < length; i++)
+            {
+                buffer[i] = (byte)((v >> (i * 8)) & 0xFF);
+            }
+            await WriteBytesAsync(GetBuffer(len), 0, length, isAsync, ct).ConfigureAwait(false);
+        }
+
+        /// <summary>
```

Comment:
Can TdsParser.SerializeDecimal be added to this Writer as well ?

Reply:
Oh, I didn't bring serializers yet, do we need all serializers as part of the writer class?

Reply:
Not all. Decimal feels like the right contender. 
Then there will be sql serializers which can leverage the tdswriter 

Don't worry about all of them. Just decimal feels like it belongs here